packages/trpc/server/routers/viewer/teams/get.handler.ts

```diff
@@ -37,8 +37,10 @@ export const getHandler = async ({ ctx, input }: GetOptions) => {
   // Hide Members of team when 1) Org is private and logged in user is not admin or owner
   // OR
   // 2)Team is private and logged in user is not admin or owner of team or Organization's admin or owner
+  const isOrgPrivate = ctx.user.profile?.organization?.isPrivate;
+  const isOrgAdminOrOwner = ctx.user.organization?.isOrgAdmin;
   const hideMembers =
-    (ctx.user.profile?.organization?.isPrivate && !ctx.user.organization?.isOrgAdmin) ||
+    (isOrgPrivate && !isOrgAdminOrOwner && input.teamId === ctx.user.organizationId) ||
```

Comment:
I think it could make more sense to convert it into a function
```javascript
function shouldHideMembers() {
	if (!team.isPrivate) {
	  return false;
	}
	return isTeamAnOrg ? !isOrgAdminOrOwner : !isOrgAdminOrOwner && !isTeamAdminOrOwner
}
```